src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -823,6 +823,26 @@ std::shared_ptr<Node> change_constant_precision<ov::element::Type_t::f16, ov::el
     return new_constant;
 }
 
+template <>
+std::shared_ptr<Node> change_constant_precision<ov::element::Type_t::f32, ov::element::Type_t::f16>(
```

Comment:
@dmitry-gorokhov, why not to share code with https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp#L17?

Reply:
great idea, could you please continue with _your idea_ in following up PR?